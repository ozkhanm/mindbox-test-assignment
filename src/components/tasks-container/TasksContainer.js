import Task from "../task/Task";

const TasksContainer = ({tasks, taskFinishButtonClickHandler}) => {
    return (
        <div>
            {tasks.map((it) => <Task key={it.id} 
                text={it.text} 
                taskStatus={it.isFinished} 
                taskFinishButtonClickHandler={taskFinishButtonClickHandler} 
                id={it.id} />)}
        </div>
    );
};

export default TasksContainer;