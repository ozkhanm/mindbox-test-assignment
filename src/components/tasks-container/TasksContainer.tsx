import React from "react";
import Task from "../task/Task";

interface TaskType {
    text: string;
    isFinished: boolean;
    id: number;
};

interface TasksContainerTypes {
    tasks: TaskType[];
    taskFinishButtonClickHandler: (id: number) => void;
};

const TasksContainer: React.FC<TasksContainerTypes> = ({tasks, taskFinishButtonClickHandler}) => {
    return (
        <div>
            {tasks.map((it: TaskType) => <Task key={it.id} 
                text={it.text} 
                taskStatus={it.isFinished} 
                taskFinishButtonClickHandler={taskFinishButtonClickHandler} 
                id={it.id} />)}
        </div>
    );
};

export default TasksContainer;