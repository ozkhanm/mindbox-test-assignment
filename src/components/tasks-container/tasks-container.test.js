import React from "react";
import renderer from "react-test-renderer";

import TasksContainer from "./TasksContainer";

it(`TasksContainer renders correctly`, () => {
    const tasks = [
        {
            text: "Тестовое задание",
            isFinished: false,
            id: 0,
        },
        {
            text: "Прекрасный код",
            isFinished: true,
            id: 1,
        },
        {
            text: "Покрытие тестами",
            isFinished: false,
            id: 2
        },
    ];
    const tree = renderer
        .create(<TasksContainer tasks={tasks} />)
        .toJSON();
  
    expect(tree).toMatchSnapshot();
});
