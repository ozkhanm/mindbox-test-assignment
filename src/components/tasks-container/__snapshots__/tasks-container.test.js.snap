// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksContainer renders correctly 1`] = `
<div>
  <div
    className="task-container small-container"
  >
    <input
      checked={false}
      className="task-status-element"
      id="task0"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className=""
      htmlFor="task0"
    >
      Тестовое задание
    </label>
  </div>
  <div
    className="task-container small-container"
  >
    <input
      checked={true}
      className="task-status-element"
      id="task1"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="task-text-finished"
      htmlFor="task1"
    >
      Прекрасный код
    </label>
  </div>
  <div
    className="task-container small-container"
  >
    <input
      checked={false}
      className="task-status-element"
      id="task2"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className=""
      htmlFor="task2"
    >
      Покрытие тестами
    </label>
  </div>
</div>
`;
